### PR TITLE
Cherry-pick #16927 to 7.x: Fix issue when building Agent snapshot and enable building the agent.

### PR DESCRIPTION
**Pull request by ph**
Cherry-pick of PR elastic/beats#16927 to 7.x branch. Original message: 

An extension in the generated files ("SNAPSHOT") was preventing us to
correctly generate the agent artifact with the packaged
metricbeat/filebeat.

Fixes: elastic/beats#16908

## Why is it important?

To have agent artifacts from the nigthly build.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: dev-tools/packaging/packages.yml [hunk 1]
```diff
@@ -98,10 +98,10 @@ shared:
         mode: 0600
         config: true
       /etc/{{.BeatName}}/beats/filebeat-{{ beat_version }}-{{.GOOS}}-{{.AgentArchName}}.tar.gz:
```
Comment (michalpristas) [inline dev-tools/packaging/packages.yml:100]: can you cherry pick also a commit where snapshot is used also here on this line?

Comment (ph) [reply dev-tools/packaging/packages.yml:100]: https://github.com/elastic/beats/pull/17077/files

This one right?

Comment (michalpristas) [reply dev-tools/packaging/packages.yml:100]: yes please

Comment (ph) [reply dev-tools/packaging/packages.yml]: I have included #17077 #17368 #17016 #16898 in this PR can you take another look?

**Comment by ph** [comment]
I am going to merge this as soon as lint passes there are currently no tests for agent executed on the 7.x branch.

**Pull request merged**